### PR TITLE
Check for SDK compatibility against current Kuzzle version

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Now Kuzzle will check the `volatile.sdkName` header to ensure that the current SDK is compatible against the current Kuzzle version.  

For Kuzzle v1 SDKs, the `sdkName` property is not present. So when the property is missing, we assume that the SDK is compatible.

For Kuzzle v2 SDKs, a compatibility matrix has been added. It works with simplified semver ranges.

Example  compat matrix:
```
{
  "js": "<7" // this version of Kuzzle is compatible with the JS SDKs prior to the 7.x.x version
}
```

Example `sdkName` volatile property: `js@7.4.2`

### CODE REVIEW

File: lib/api/controllers/funnelController.js [hunk 1]
```diff
@@ -679,8 +728,8 @@ class FunnelController {
  * @param {string} string
  * @returns {string}
  */
-function capitalize(string) {
-  return string.charAt(0).toUpperCase() + string.slice(1);
+function capitalize (string) {
+  return string[0].toUpperCase() + string.substr(1);
```
Comment (Aschen) [inline lib/api/controllers/funnelController.js]:
```
CharAt Uppercase + slice x 18,950,356 ops/sec ±0.60% (94 runs sampled)
CharAt Uppercase + substr x 19,256,026 ops/sec ±0.39% (93 runs sampled)
string[] Uppercase + substr x 49,175,951 ops/sec ±0.68% (94 runs sampled)
```

File: lib/api/controllers/funnelController.js [hunk 2]
```diff
@@ -709,4 +758,22 @@ function doAction(controller, request) {
...
+/**
+ * Very straightforward function to check only if the version satisfies
+ * semver range major version
+ *
+ * @param {String} version
+ * @param {String} range
+ *
+ * @returns {Boolean}
+ */
+function satisfiesMajor (version, range) {
```
Comment (Aschen) [inline lib/api/controllers/funnelController.js]: Semver is very slow and we didn't need a complete range check. (this check has to be performed for every request)
```
Semver x 186,166 ops/sec ±4.04% (90 runs sampled)
Manual: only major x 683,741,550 ops/sec ±0.52% (82 runs sampled)
```

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L918](https://github.com/kuzzleio/kuzzle/blob/5f0d5e42805921380d1624d67137fdc06e5750bd/lib/api/core/plugins/pluginsManager.js#L918): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: test/api/controllers/funnelController/processRequest.test.js [hunk 3]
```diff
@@ -133,6 +135,17 @@ describe('funnelController.processRequest', () => {
       });
   });
 
+  it('should rejects if _checkSdkVersion fail', () => {
```
Comment (scottinet) [inline test/api/controllers/funnelController/processRequest.test.js]:
```suggestion
  it('should reject if _checkSdkVersion fails', () => {
```

**Pull request merged**